tensorflow/stream_executor/rocm/rocm_driver.cc

```diff
@@ -1421,11 +1272,11 @@ static port::StatusOr<T> GetSimpleAttribute(hipDevice_t device,
   return max_blocks;
 }
 
-/* static */ hipCtx_t ROCMDriver::CurrentContextOrDie() {
-  hipCtx_t current = nullptr;
-  hipError_t result = hipCtxGetCurrent(&current);
+/* static */ int ROCMDriver::CurrentDeviceOrDie() {
+  int current = -1;
+  hipError_t result = hipGetDevice(&current);
```

Comment:
@deven-amd Thanks for this contribution!

From the look of the PR the crux of the proposed change is to abolish usage of `hipCtx` and depends on `hipGetDevice()` to understand the current GPU device the current thread is operating on, correct?

Reply:
My next question is that in `tensorflow/core/common_runtime/gpu/gpu_device.cc` there are notions of `TF GPU ID` and `Physical GPU ID`. Do we need to bother adopting those notions? `hipGetDevice()` should give us `Physical GPU ID` I think.

Reply:
Correct on the first one.

On the second one, I need to revisit the code and ensure that I am not mixing up the two. I will post an update tomorrow after looking at the code.

Reply:
@deven-amd The test is a success. I'd like to hear from you regarding to `TF GPU ID` and `Physical GPU ID` before taking the PR.

Reply:
I think for the code in this PR, we are only concerned with the `Physical GPU ID` 

The GPU ID is passed in via the `StreamExecutor::Init` `-->` `ROCMExecutor::Init` routine, we cache it as the `device_ordinal_` member of `ROCMExecutor` , and then use that id / number for all `HIP` API calls associated with that executor.

The `ScopedActivateContext` (I did not change the name to `ScopedActivateDevice` as that would have triggered some more name updates) is used to ensure that the "correct" GPU is current before making calls to `HIP`API. The switch to the correct GPU (if necessary) happens in the constructor and the switch-back (again if necessary) in the destructor. You will notice instantiations of `ScopedActivateContext` in many of the routines in `rocm_driver.cc`